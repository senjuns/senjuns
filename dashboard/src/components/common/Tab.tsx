import React from 'react';
import styled from 'styled-components';
import { Colors } from '../../shared/constants';

interface TabProps {
  selected: boolean;
  label: string;
  onClick: () => void;
  id: string;
}

const Tab: React.FC<TabProps> = ({
  selected,
  label,
  onClick,
  id,
}: TabProps) => {
  const labelStyle = {
    color: selected ? Colors.orange2 : Colors.dark7,
    fontWeight: selected ? 600 : 400,
  };

  return (
    <Container data-selected={selected} id={id}>
      <Label style={labelStyle} onClick={onClick}>
        {label}
      </Label>
      {selected && (
        <CircleContainer>
          <Circle />
        </CircleContainer>
      )}
    </Container>
  );
};

const Container = styled.div`
  width: fit-content;
`;

const Label = styled.div`
  font-family: Poppins;
  font-size: 20px;
  height: 30px;
  cursor: pointer;
  white-space: nowrap;
`;

const Circle = styled.div`
  width: 8px;
  height: 8px;
  border-radius: 4px;
  background-color: #e86339;
`;

const CircleContainer = styled.div`
  display: flex;
  justify-content: center;
`;

export default Tab;
