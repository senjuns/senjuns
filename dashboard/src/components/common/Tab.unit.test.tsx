// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';
import Tab from './Tab';

describe('<Tab />', () => {
  it('renders correctly.', async () => {
    const handleClick = jest.fn();
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <Tab id={'test_tab'} selected={true} label="test" onClick={handleClick} />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
