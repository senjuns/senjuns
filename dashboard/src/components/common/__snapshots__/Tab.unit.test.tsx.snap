// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> renders correctly. 1`] = `
<styled.div
  data-selected={true}
  id="test_tab"
>
  <styled.div
    onClick={[MockFunction]}
    style={
      Object {
        "color": "#E86339",
        "fontWeight": 600,
      }
    }
  >
    test
  </styled.div>
  <styled.div>
    <styled.div />
  </styled.div>
</styled.div>
`;
